### DISCUSSION TITLE
What is the use of .refit_full()?

### DISCUSSION THREAD

**Discussion by wschung1113** [autogluon/autogluon#4016] [Q&A]
Hi everyone,

Posting this because I'm not sure of the purpose of .refit_full().

Fyi, I am referring to time series predictor models here.

As of my understanding, .refit_full() re-trains (as in updating the trained model with extra data) the trained predictors with the FULL train data including the validation set that was left out for metrics evaluation.

In the default case, autogluon-ts leaves out the last chunk length of prediction horizon of the training data provided for validation.

After training is done, when .refit_full() method is called, models are trained to fit even the previously left-out validation set.

So if there is no intention for hyperparameter tuning, is training of predictors done with a left-out validation set solely to rank which models performed best?

Thanks for reading,
have a good day.

**Comment by shchur** [comment]
Hi @wschung1113, thank you for the question. Yes, that's exactly the intention. After we perform model selection/ensemble fitting, we can refit the base models using all the available data. Note that not all models need this: for example, local models like ARIMA are automatically refit at prediction time, so we don't refit them. Similarly, deep learning models like DeepAR really benefit from having a validation set, so we skip refit for them as well.

Also note that in AutoGluon v1.0 there is a bug that breaks this functionality for time series (fixed in #https://github.com/autogluon/autogluon/pull/3820), the fix should be available soon in the v1.1 release.

**Comment by atayeby** [comment]
Is there an equivalent where we can refit_full, where the full dataset is used including any data that was put aside for the test dataset?

I'm fairly new to datascience and time series forecasting. I'm using a test and train dataset, but want to make sure that the most recent datapoints, which to me seem like very important data are included in the final forecasts.

**Reply by yaniv92648** [reply]
I think refit_full=True in the fit method of TimeSeriesPredictor - does this.
Am I right?
@shchur
